### PR TITLE
Split dns_packet_test.exs along feature lines

### PR DESCRIPTION
**Pull request by toshi0806**
Resolves #96

## Summary

Pure test reorganization — zero production changes. The 3,145-line `dns_packet_test.exs` is split along feature lines, adjusted to what actually remains after #94 moved the per-type rdata tests to the round-trip suite:

| File | Lines | Content |
|---|---|---|
| `test/dns_packet_edns_internal_test.exs` (new) | 2,133 | All 14 EDNS describes: `parse_edns_info/1`, ECS edge cases, option creation/parsing, `merge_edns_info_to_additional`, integration tests. Internal-contract moduledoc pointing to the round-trip suites as the compatibility contract. |
| `test/dns_packet_name_test.exs` (new) | 231 | Domain-name encoding, character strings, and name-compression tests against crafted binaries (pointer-following that `create/1` never produces). |
| `test/dns_packet_test.exs` | 3,145 → 808 | Rdata fallback contracts, legacy round-trip describe, additional coverage, advanced record types, SVCB parameter tests. Keeps the internal-contract charter moduledoc from #94. |

## Identity gates (as required by the issue)

- `mix test`: exactly **221 passed** before and after — no test added, dropped or altered
- Coverage: unchanged at **93.81%**
- `mix credo --strict`: clean
- Only `test/` touched; `git diff --stat` shows 3 files, no `lib/` changes

### CODE REVIEW

File: test/dns_packet_edns_internal_test.exs [hunk 1]
```diff
@@ -0,0 +1,2133 @@
...
+        }
+      }
+
+      binary = DNSpacket.create(packet)
+      parsed = DNSpacket.parse(binary)
+
+      assert parsed.id == 0x2222
+      # Just check that packet was parsed successfully
+      assert is_struct(parsed, DNSpacket)
+    end
```
[the remark this answers is not in the document]
Comment (toshi0806) [reply test/dns_packet_edns_internal_test.exs:1958]: Verified — correct on all three counts: `chain_point_of_trust` and `edns_key_tag_list` are not canonical flat keys (`chain_closest_encloser` / `edns_key_tag_key_tags`), so `unflatten/1` drops them and those tests assert nothing beyond parse success.

Not fixed in this PR by design: its whole verifiability rests on the identity gate (exactly 221 tests, nothing altered), so content changes don't belong here. Filed as #101 with the analysis — note the canonical-key round-trips for `chain` / `edns_key_tag` / `edns_expire` already exist in `dns_packet_edns_consistency_test.exs`, so the likely fix is deleting the no-op tests rather than rewriting them.
